lua/gitlab/utils/init.lua

```diff
@@ -656,6 +695,27 @@ M.get_all_git_branches = function(remote)
   return branches
 end
 
+---Select a git branch, add it to callback args and perform callback
+---@param cb function The call back to perform with the selected branch
+---@param args table Table with callback arguments
+---@param arg string Name under which the selected branch will be added to the args
+---@param remote? boolean Whether remote-tracking branches should be listed
+---@param exclude_current? boolean Whether the current branch should be excluded from the list
+M.select_target_branch = function(cb, args, arg, remote, exclude_current)
```

Comment:
We are only supporting opening up MRs against branches that are on the remote since that's all that Gitlab can do, this function doesn't need any arguments. Please also look into using the `List` module in the future for logic like this, it makes things easier to understand. I'm trying to get away from so many C-style for loops which are difficult to follow.

Reply:
I think this is all that's needed here, no need for all the `args` and booleans:

```lua
M.get_all_merge_targets = function()
  local handle = io.popen("git branch -r 2>&1")
  if not handle then
    M.notify("Error running 'git branch' command.", vim.log.levels.ERROR)
    return
  end

  local current_branch = M.get_current_branch()
  if not current_branch then
    return
  end

  local lines = {}
  for line in handle:lines() do
    table.insert(lines, line)
  end
  handle:close()

  -- Trim "origin/" and don't include the HEAD pointer
  local branches = List.new(lines)
      :map(function(line)
        return line:match("origin/(%S+)")
      end)
      :filter(function(branch)
        return not branch:match("^HEAD$") and branch ~= current_branch
      end)

  return branches
end

---Select a git branch
---@param cb function The call back to perform with the selected branch
M.select_target_branch = function(cb)
  local all_branch_names = M.get_all_merge_targets()
  if not all_branch_names then
    return
  end
  vim.ui.select(all_branch_names, {
    prompt = "Choose target branch for merge",
  }, function(choice)
    if choice then
      cb(choice)
    end
  end)
end
```

Reply:
Hi Harrison, thanks for your suggestions. It didn't occur to me to use the `List` functionality but it will certainly make things clearer. I wanted to keep the `remote` parameter and introduced the `exclude_current` parameter in `get_all_git_branches` to really make it possible to get *all* git branches. But as this is not really necessary for the plugin, I'll remove those two parameters.

I'm not sure, however, about the `args` parameter. The reason I used it is that the `utils.select_target_branch` method replaces this piece of code:
```lua
  local all_branch_names = u.get_all_git_branches(true, true)
  vim.ui.select(all_branch_names, {
    prompt = "Choose target branch for merge",
  }, function(choice)
    if choice then
      mr.target = choice
      M.pick_template(mr)
    end
  end)
```
As you can see, the `choice` of `vim.ui.select` was not actually passed to the `callback` method (`create_mr.pick_template`) directly, but it was only added to the other arguments, i.e., `mr = {target = "THE PICKED TARGET", delete_branch = true, ...}`. I'll try to figure something out.

Reply:
I've made the `Delete branch`, `Squash commits` and `Target branch` fields non-modifiable to prevent users from inserting invalid values. I was also thinking about adding an autocommand that would show help if a user attempted to modify the buffers, something like:
```lua
  vim.api.nvim_create_autocmd("CursorHold", {
    buffer = buf,
    callback = function()
      if vim.v.errmsg == "E21: Cannot make changes, 'modifiable' is off" then
        M.notify("Do not edit manually. Use " .. require("gitlab.state").settings.popup.perform_linewise_action .. " instead.", vim.log.levels.INFO)
        vim.v.errmsg = ""
      end
    end,
  })
```
But that's probably superfluous.
Now I'm rather considering whether to simplify the `toggle_string_bool` method in `lua/gitlab/utils/init.lua:377`, since it does not need all the case and whitespace checking anymore.

Reply:
Looks like you requested another review but I don't think you pushed up your changes